harvester_e2e_tests/fixtures/terraform_rancher.py

```diff
@@ -0,0 +1,349 @@
+      config = {
+        cloud-provider-config = file("${path.module}/kubeconfig")
+        cloud-provider-name = "harvester"
+      }
+    }
+    machine_global_config = <<EOF
+    cni: "calico"
+    disable-kube-proxy: false
+    etcd-expose-metrics: false
+    EOF
+    upgrade_strategy {
+      control_plane_concurrency = "10%%"
+      worker_concurrency        = "10%%"
+    }
+    etcd {
+      snapshot_schedule_cron = "0 */5 * * *"
+      snapshot_retention     = 5
+    }
+    chart_values = <<EOF
+    harvester-cloud-provider:
+      clusterName: %(harvester_name)s
+      cloudConfigPath: /var/lib/rancher/rke2/etc/config-files/cloud-provider-config
+    EOF
+}
+'''
+
+TF_MACHINE_POOLS = '''
+    machine_pools {
+      name                         = "pool1"
+      cloud_credential_secret_name = rancher2_cloud_credential.%(cloud_credential_name)s.id
+      control_plane_role           = true
+      etcd_role                    = true
+      worker_role                  = true
+      quantity                     = 1
+      machine_config {
+        kind = rancher2_machine_config_v2.%(machine_config_name)s.kind
+        name = rancher2_machine_config_v2.%(machine_config_name)s.name
+      }
+    }
+'''
+
+
+def remove_ansicode(ctx):
+    if isinstance(ctx, bytes):
+        ctx = ctx.decode()
+    return re.sub(r"\x1b|\[\d+m", "", ctx)
+
+
+@pytest.fixture(scope="session")
+def tf_script_dir(request):
+    return Path(request.config.getoption('--terraform-scripts-location'))
+
+
+@pytest.fixture(scope="session")
+def tf_provider_version(request):
+    version = request.config.getoption('--terraform-provider-rancher')
+    if not version:
+        import requests
+        resp = requests.get("https://registry.terraform.io/v1/providers/rancher/rancher2")
+        version = max(resp.json()['versions'], key=parse_version)
+    return version
+
+
+@pytest.fixture(scope="session")
+def tf_executor(tf_script_dir):
+    run(str(tf_script_dir / "terraform_install.sh"), stdout=PIPE, stderr=PIPE)
+    executor = tf_script_dir / "bin/terraform"
+    assert executor.is_file()
+
+    yield executor
+
+
+@pytest.fixture(scope="module")
+def tf_rancher(api_client, tf_script_dir, tf_provider_version, tf_executor, harvester, rancher):
+    tf_rancher = TerraformRancher(tf_executor,
+                                  tf_script_dir / datetime.now().strftime("%Hh%Mm_%m-%d"))
+
+    # kubeconfig for rancher provider
+    # Ref. https://registry.terraform.io/providers/rancher/rancher2/3.1.1/docs/resources/cluster_v2
+    url = f'{rancher["endpoint"]}/k8s/clusters/{harvester["id"]}/v1/harvester/kubeconfig'
+    auth = (rancher["access_key"], rancher["secret_key"])
+    data = {
+        "clusterRoleName": "harvesterhci.io:cloudprovider",
+        "namespace": "default",
+        "serviceAccountName": harvester["name"]
+    }
+    res = requests.post(url, auth=auth, json=data, verify=False)
+    kubeconfig = res.text.replace("\\n", "\n").strip('"')
```

Comment:
the part to get kubeconfig from Rancher could move to `RancherAPI.generate_kubeconfig()` like HarvesterAPI did.